### PR TITLE
Sort collections in 'Add to Collection' dropdown alphabetically (bug 1041234)

### PR DESCRIPTION
**Pull request by zuzelvp**
…1041234).

https://bugzilla.mozilla.org/show_bug.cgi?id=593634

Reproduced:

reproduced

Fixed:

fixed

### CODE REVIEW

File: apps/bandwagon/models.py [hunk 1]
```diff
@@ -65,9 +65,10 @@ def listed(self):
 
     def publishable_by(self, user):
         """Collections that are publishable by a user."""
-        owned_by = list(self.filter(author=user.id))
-        publishable_by = list(self.filter(users=user.id))
-        return set(owned_by + publishable_by)
+        owned_by = models.Q(author=user.id)
+        publishable_by = models.Q(users=user.id)
+        collections = self.filter(owned_by | publishable_by)
```
Comment (magopian) [inline apps/bandwagon/models.py]: Very clever, I like that ;)

File: apps/bandwagon/tests/test_models.py [hunk 2]
```diff
@@ -138,6 +138,17 @@ def test_publishable_by(self):
...
         eq_(c.publishable_by(self.user), True)
 
+    def test_manager_publishable_by(self):
+        c1 = Collection.objects.create(author=self.user, name='B')
+        c2 = Collection.objects.create(author=self.user, name='A')
+        c3 = Collection.objects.create(author=self.other, name='D')
+        Collection.objects.create(author=self.other, name='C')
+        CollectionUser(collection=c1, user=self.user).save()
+        CollectionUser(collection=c2, user=self.other).save()
+        CollectionUser(collection=c3, user=self.user).save()
```
Comment (magopian) [inline apps/bandwagon/tests/test_models.py]: For good measure, we should also have a `CollectionUser` for the "other" user here, just to make sure that even if there's one, it still doesn't return it in the publishable list.

File: apps/bandwagon/tests/test_models.py [hunk 3]
```diff
@@ -138,6 +138,17 @@ def test_publishable_by(self):
...
+        collections = Collection.objects.publishable_by(self.user)
+        eq_([c.name for c in collections], [c2.name, c1.name, c3.name])
```
Comment (magopian) [inline apps/bandwagon/tests/test_models.py]: We use [pytest](https://pytest.org) for tests now, so even if most of the tests are still written using `eq_`, we rather use simple `assert`s now. Also, I think we can compare collections themselves, no need to compare their names:
`assert list(collections) == [c2, c1, c3]`

**Comment by magopian** [comment]
This is excellent, thanks! Two minor nits, and it'll be good to merge

**Comment by magopian** [comment]
Also, if you could squash those commits at the end, that would be helpful, otherwise i'll just do it when merging.

File: apps/bandwagon/tests/test_models.py (hunk 2 above)
Comment (zuzelvp) [inline apps/bandwagon/tests/test_models.py]: Will fix.

File: apps/bandwagon/tests/test_models.py (hunk 3 above)
Comment (zuzelvp) [inline apps/bandwagon/tests/test_models.py]: Ok.

**Comment by magopian** [comment]
Perfect! Thanks a lot for this contribution!
r+

**Comment by magopian** [comment]
**/!\ The bug number in the commit is wrong, the correct bug number is 593634 /!\**

**Pull request merged**